engine/crates/engine-v2/src/sources/graphql/mod.rs

```diff
@@ -84,6 +85,15 @@ pub(crate) struct GraphqlExecutor<'ctx> {
 impl<'ctx> GraphqlExecutor<'ctx> {
     #[tracing::instrument(skip_all, fields(plan_id = %self.plan.id(), federated_subgraph = %self.subgraph.name()))]
     pub async fn execute(mut self) -> ExecutionResult<ResponsePart> {
+        let operation_name = self.plan.operation().as_ref().name.as_ref();
+        let operation_type = self.plan.operation().as_ref().ty.as_ref();
+
+        let subgraph_request_span = grafbase_tracing::span::subgraph::SubgraphRequestSpan::new(self.subgraph.name())
+            .with_operation_type(operation_type)
+            .with_operation_name(operation_name.map(|s| s.as_str()))
+            .with_document(self.json_body.as_str())
```

Comment:
operation name isn't useful here, it's the one for the root request. We don't generate operation names for sub requests anymore.
For the document, I'm mitigated. you're including variables in the `json_body` which is something we shouldn't keep around. The query string makes more sense. That's something that could be kept separated in the `GraphqlExecutor`. So would add a `FIXME:` here and use an empty string for the document. Can do it later

Reply:
The spans and their attributes follow the Apollo docs. We can discuss it from a product perspective but as is, anything graphql related should be in the graphql span which is for that purpose - the root is more for the protocol.
you're 100% right on the variables, we shouldn't be considering those.

Reply:
Added the `FIXME:` and removed it from the time being.